exe/main.cc

```diff
@@ -40,90 +53,118 @@ int startAgent(std::vector<std::string> arguments) {
         }
     } catch (const std::exception& e) {
         LOG_ERROR("Failed to daemonize: %1%", e.what());
-        return 1;
+        return EXIT_FAILURE;
     } catch (...) {
         LOG_ERROR("Failed to daemonize");
-        return 1;
-    }
-
-    if (!Configuration::Instance().isInitialized()) {
-        // Start a thread that just busy waits to facilitate
-        // acceptance testing
-        PCPClient::Util::thread idle_thread {
-            []() {
-                LOG_WARNING("pxp-agent started unconfigured. No connection can "
-                            "be established");
-                for (;;) {
-                    PCPClient::Util::this_thread::sleep_for(PCPClient::Util::chrono::milliseconds(5000));
-                }
-            } };
-        idle_thread.join();
-        return 0;
+        return EXIT_FAILURE;
     }
 
-    bool success { false };
-    const auto& agent_configuration =
-        Configuration::Instance().getAgentConfiguration();
-
-    try {
-        Agent agent { agent_configuration };
-        agent.start();
-        success = true;
-    } catch (const Agent::Error& e) {
-        LOG_ERROR("Fatal error: %1%", e.what());
-    } catch (const std::exception& e) {
-        LOG_ERROR("Unexpected error: %1%", e.what());
-    } catch (...) {
-        LOG_ERROR("Unexpected error");
+    bool success { true };
+    if (!Configuration::Instance().valid()) {
+        // pxp-agent will execute in uncofigured mode
+        loopIdly();
+    } else {
+        try {
+            Agent agent { Configuration::Instance().getAgentConfiguration() };
+            agent.start();
+        } catch (const Agent::WebSocketConfigurationError& e) {
```

Comment:
New. The underlying WebSocket TLS init handle will throw an exception in case it fails to load the cert / key pair, which  the Agent instance will propagate in that way.